project-base/storefront/components/Blocks/SocialNetworkLogin/SocialNetworkLogin.tsx

```diff
@@ -0,0 +1,52 @@
+import { SocialNetworkLoginLink } from './SocialNetworkLoginLink';
+import { TypeLoginTypeEnum } from 'graphql/types';
+import React from 'react';
+import { usePersistStore } from 'store/usePersistStore';
+
+type SocialNetworkLoginProps = {
+    socialNetworks: TypeLoginTypeEnum[];
+    shouldOverwriteCustomerUserCart: boolean | undefined;
+};
+
+export const SocialNetworkLogin: FC<SocialNetworkLoginProps> = ({
+    socialNetworks,
+    shouldOverwriteCustomerUserCart,
+}) => {
+    const cartUuid = usePersistStore((store) => store.cartUuid);
+    const productListUuids: string[] = Object.values(usePersistStore((store) => store.productListUuids));
+
+    const getSocialNetworkLoginUrl = (
+        socialNetwork: TypeLoginTypeEnum,
+        cartUuid: string | null,
+        shouldOverwriteCustomerUserCart: boolean | undefined,
+        productListUuids: string[],
+    ) => {
+        let url = `/social-network/login/${socialNetwork}`;
+        if (cartUuid) {
+            url += `?cartUuid=${cartUuid}&shouldOverwriteCustomerUserCart=${shouldOverwriteCustomerUserCart ? 'true' : 'false'}`;
+        }
+        if (productListUuids.length > 0) {
+            const separator = cartUuid ? '&' : '?';
+            url += `${separator}productListUuids=${productListUuids.join(',')}`;
+        }
+
+        return url;
+    };
```

Comment:
I know you were just moving this code around, but isn't there an easier way how to build url for example using `URL` of even passing `href` as object rather then string (as described in the [nextjs docs](https://nextjs.org/docs/pages/api-reference/components/link#href-required)).

Reply:
fixed